subprojects/workers/src/main/java/org/gradle/workers/internal/DefaultWorkerExecutor.java

```diff
@@ -153,6 +170,40 @@ DaemonForkOptions getDaemonForkOptions(Class<?> actionClass, WorkerConfiguration
         return toDaemonOptions(actionClass, paramTypes, configuration.getForkOptions(), configuration.getClasspath());
     }
 
+    private void validateWorkerConfiguration(WorkerConfiguration configuration) {
```

Comment:
:o: I wonder if we could move this code to `IsolationMode` itself. Like: `IsolationMode#isCompatibleWith(WorkerConfiguration config)`.